src/components/navigation-control.js

```diff
@@ -42,7 +47,7 @@ export default class NavigationControl extends BaseControl {
 
   _updateViewport(opts) {
     const {viewport} = this.context;
-    const mapState = new MapState(Object.assign({}, viewport, opts));
+    const mapState = new MapState(Object.assign({}, viewport, LINEAR_TRANSITION_PROPS, opts));
```

Comment:
`LINEAR_TRANSITION_PROPS` has `transitionDuration` set, is it intentional to enable transition in this flow?

Reply:
Yes, this is trying to mimic Mapbox 's behavior